src/WebJobs.Script/Rpc/WebHostLanguageWorkerChannelManager.cs

```diff
@@ -37,8 +39,13 @@ public WebHostLanguageWorkerChannelManager(IScriptEventManager eventManager, IEn
             _logger = loggerFactory.CreateLogger<WebHostLanguageWorkerChannelManager>();
             _applicationHostOptions = applicationHostOptions;
 
+            _workerRestartSubscription = _eventManager.OfType<WorkerShutdownEvent>()
```

Comment:
nit -- should this be called `_workerShutdownSubscription`?

Reply:
Thinking more about this, we should introduce 'watchDirectories` in worker.config.json that lists directories  which a language worker  can specify so that it can opt in to which file changes should restart a worker. 
So this subscription should restart worker on changes to:
- Any file extensions that language worker supports for e.g node worker will restart on changes to `.js` files
- Any directories listed under watch directories. For e.g: node worker should restart if any thing changes in folder `node_modules`